notie.js

```diff
@@ -11,699 +11,799 @@
-    var input_yes_id = 'notie-input-yes';
-    var input_no_id = 'notie-input-no';
-    var input_text_id = 'notie-input-text';
-    var input_yes_text_id = 'notie-input-yes-text';
-    var input_no_text_id = 'notie-input-no-text';
-    
-    // *********************************************
-    
-    
-    
-    // NOTIE.ALERT
-    // *********************************************
-
-    // notie elements and styling
-    var alert_outer = document.createElement('div');
-    alert_outer.id = alert_outer_id;
-    alert_outer.style.position = 'fixed';
-    alert_outer.style.top = '0';
-    alert_outer.style.left = '0';
-    alert_outer.style.zIndex = '999999999';
-    alert_outer.style.height = 'auto';
-    alert_outer.style.width = '100%';
-    alert_outer.style.display = 'none';
-    alert_outer.style.textAlign = 'center';
-    alert_outer.style.cursor = 'default';
-    alert_outer.style.MozTransition = '';
-    alert_outer.style.WebkitTransition = '';
-    alert_outer.style.transition = '';
-    alert_outer.style.cursor = 'pointer';
-    alert_outer.addEventListener('click', function() {
-        clearTimeout(alert_timeout_1);
-        clearTimeout(alert_timeout_2);
-        alert_hide();
-    });
-    var alert_inner = document.createElement('div');
-    alert_inner.id = alert_inner_id;
-    alert_inner.style.padding = '20px';
-    alert_inner.style.display = 'table-cell';
-    alert_inner.style.verticalAlign = 'middle';
-    alert_outer.appendChild(alert_inner);
-
-    // Initialize notie text
-    var alert_text = document.createElement('span');
-    alert_text.id = alert_text_id;
-    alert_text.style.color = alert_color_text;
-    if (window.innerWidth <= font_change_screen_width) { alert_text.style.fontSize = font_size_small; }
-    else { alert_text.style.fontSize = font_size_big; }
-    window.addEventListener('resize', function(){
-        if (window.innerWidth <= font_change_screen_width) { alert_text.style.fontSize = font_size_small; }
-        else { alert_text.style.fontSize = font_size_big; }
-    }, true);
-    alert_inner.appendChild(alert_text);
-
-    // Attach notie to the body element
-    document.body.appendChild(alert_outer);
-
-    // Declare variables
-    var height = 0;
-    var alert_is_showing = false;
-    var alert_timeout_1;
-    var alert_timeout_2;
-    var was_clicked_counter = 0;
-
-    function alert(type, message, seconds) {
-
-        was_clicked_counter++;
-
-        setTimeout(function() {
-            was_clicked_counter--;
-        }, (animation_delay * 1000 + 10));
-
-        if (was_clicked_counter == 1) {
+ */
+/* jshint strict:true, noempty:true, noarg:true, eqeqeq:true, browser:true, bitwise:true, curly:true, undef:true, nonew:true, forin:true */
+/* global define: true, exports: true, require: true, module: true */
+(function (root, factory) {
+    'use strict';
+    if (typeof define === 'function' && define.amd) {
+        // AMD. Register as an anonymous module.
+        define([], factory);
```

Comment:
Mmm, now that you say yes... the times I've used AMD I always end referencing the file where the module is. Also their own documentation stating to avoid naming modules gives +1 to not name the module. Restored.